src/lib/utils/scales-utils.js

```diff
@@ -78,24 +78,26 @@ const SCALE_FUNCTIONS = {
 /**
  * Find the smallest distance between the values on a given scale and return
  * the index of the element, where the smallest distance was found.
+ * It returns the first occurrence of i where
+ * `scale(value[i]) - scale(value[i - 1])` is minimal
  * @param {Array} values Array of values.
  * @param {Object} scaleObject Scale object.
  * @returns {number} Index of an element where the smallest distance was found.
  * @private
  */
-function _getSmallestDistanceIndex(values, scaleObject) {
+export function _getSmallestDistanceIndex(values, scaleObject) {
```

Comment:
I think we should strive for correctness and stability. I've seen private APIs in the wild, they're usually marked with underscore. If users choose to rely on these methods, we make no guarantees, and it's indicated here that it's private API.